### PR TITLE
no warn if `pathMatch` is empty (fix #3072)

### LINKED ISSUE #3072: Named catch-all route warns if `pathMatch` is empty
### Version
3.1.3

### Reproduction link
[http://jsfiddle.net/yza5k02v/](http://jsfiddle.net/yza5k02v/)

### Steps to reproduce
- click on /nope
- click on button

### What is expected?
no warn

### What is actually happening?
there is a warn

---
maybe the param should be marked as optional to prevent this warning from ever happening

### PR DESCRIPTION
**Pull request by yinm**
Try to fix https://github.com/vuejs/vue-router/issues/3072
if pathMatch is empty, skip warning.

### CODE REVIEW

File: src/util/params.js [hunk 1]
```diff
@@ -24,7 +24,10 @@ export function fillParams (
 
     return filler(params, { pretty: true })
   } catch (e) {
-    if (process.env.NODE_ENV !== 'production') {
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      params.pathMatch !== '' // Fix #3072 no warn if `pathMatch` is empty
```
Comment (posva) [inline src/util/params.js]: let's check if typeof is not string on the warn method (replacing the `false` value)

**Review by posva — CHANGES_REQUESTED** [review]

File: src/util/params.js (hunk 1 above)
Comment (yinm) [reply src/util/params.js]: fixed at https://github.com/vuejs/vue-router/pull/3081/commits/590adbbf1e78662e58fc26ee656ea600261db1c4.
I also simplify source code at https://github.com/vuejs/vue-router/pull/3081/commits/8a213d9f49f0f22b858672879b5077130a3a3db4.

File: src/util/params.js [hunk 2]
```diff
@@ -24,9 +24,8 @@ export function fillParams (
 
     return filler(params, { pretty: true })
   } catch (e) {
-    if (process.env.NODE_ENV !== 'production') {
```
Comment (posva) [inline src/util/params.js:27]: the if was necessary though

**Review by posva — CHANGES_REQUESTED** [review]

File: src/util/params.js (hunk 2 above)
Comment (yinm) [reply src/util/params.js:27]: oops, I revert it

**Review by posva — APPROVED** [review]
Thanks!

**Pull request merged**